libtsuba/src/WriteGroup.cpp

```diff
@@ -25,49 +25,74 @@ WriteGroup::Make() {
   return std::unique_ptr<WriteGroup>(new WriteGroup(tag));
 }
 
+bool
+WriteGroup::Drain() {
+  auto op_it = pending_ops_.begin();
+  if (op_it == pending_ops_.end()) {
+    return false;
+  }
+  auto res = op_it->result.get();
+  if (!res) {
+    KATANA_LOG_DEBUG(
```

Comment:
This waits for writes in the order they were written?  Probably fine for now.  You could imagine a non-blocking check that found if any write had completed.